source/ios/tools/IOSFeed/IOSFeed.cs

```diff
@@ -0,0 +1,146 @@
+
+        private static async Task ProcessAsync()
+        {
+            string storageConnectionString;
+
+            FileInfo fi = new FileInfo(Constants.ConnectionStringPath);
+            if (!fi.Exists)
+            {
+                throw new InvalidOperationException("connection string file does not exist");
+            }
+
+            using (StreamReader sr = File.OpenText(Constants.ConnectionStringPath))
+            {
+                storageConnectionString = sr.ReadToEnd();
+            }
+
+            // Check whether the connection string can be parsed.
+            CloudStorageAccount storageAccount;
+            if (CloudStorageAccount.TryParse(storageConnectionString, out storageAccount))
+            {
+                // If the connection string is valid, proceed with operations against Blob
+                // storage here.
+                // Create the CloudBlobClient that represents the 
+                // Blob storage endpoint for the storage account.
+                CloudBlobClient cloudBlobClient = storageAccount.CreateCloudBlobClient();
+
+                CloudBlobContainer cloudBlobContainer =
+                    cloudBlobClient.GetContainerReference(Constants.ContainerId);
+                await cloudBlobContainer.CreateIfNotExistsAsync();
+
+                // Set the permissions so the blobs are public.
+                BlobContainerPermissions permissions = new BlobContainerPermissions
+                {
+                    PublicAccess = BlobContainerPublicAccessType.Blob
+                };
+
+                await cloudBlobContainer.SetPermissionsAsync(permissions);
+
+                var sourceFile = Path.Combine(Constants.FrameworkPath, Constants.FrameworkName);
+                var blobGuid = Guid.NewGuid().ToString();
+                var cloudFileName = blobGuid + Constants.FrameworkName;
+
+                CloudBlockBlob cloudBlockBlob = cloudBlobContainer.GetBlockBlobReference(cloudFileName);
+
+                await cloudBlockBlob.UploadFromFileAsync(sourceFile);
+
+                BlobContinuationToken blobContinuationToken = null;
+                do
+                {
+                    var results = await cloudBlobContainer.ListBlobsSegmentedAsync(null, blobContinuationToken);
+                    // Get the value of the continuation token returned by the listing call.
+                    blobContinuationToken = results.ContinuationToken;
+                    foreach (IListBlobItem item in results.Results)
+                    {
+                        var uriString = item.Uri.ToString();
+                        if (uriString.Contains(blobGuid))
+                        {
+                            UpdatePodSpec(uriString);
+                        }
+                    }
+                } while (blobContinuationToken != null);
+            }
+            else
+            {
+                throw new InvalidOperationException("invalid connection string");
+            }
```

Comment:
NIt Nit: maybe have this be part of the if check upfront for readability and move the bigger logic behind it in the else; Also include the connection string that we are complaining about for debugging.

Reply:
thanks, that's good suggestion. but i have my doubt about including the connection string in the error message. 

---
In reply to: [314076636](https://github.com/microsoft/AdaptiveCards/pull/3291#discussion_r314076636) [](ancestors = 314076636)